anomdetect/src/main/java/com/expedia/adaptivealerting/anomdetect/forecast/point/algo/sma/SmaPointForecaster.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2018-2019 Expedia Group, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.expedia.adaptivealerting.anomdetect.forecast.point.algo.sma;
+
+import com.expedia.adaptivealerting.anomdetect.forecast.point.PointForecast;
+import com.expedia.adaptivealerting.anomdetect.forecast.point.PointForecaster;
+import com.expedia.metrics.MetricData;
+import com.google.common.collect.EvictingQueue;
+
+import lombok.Generated;
+import lombok.Getter;
+
+import static com.expedia.adaptivealerting.anomdetect.util.AssertUtil.notNull;
+
+/**
+ * Point forecaster based on the Simple Moving Average (SMA) method
+ */
+public class SmaPointForecaster implements PointForecaster {
+
+    @Getter
+    @Generated // https://reflectoring.io/100-percent-test-coverage/
+    private SmaPointForecasterParams params;
+
+    @Getter
+    @Generated // https://reflectoring.io/100-percent-test-coverage/
+    private double mean;
+
+    @Getter
+    @Generated // https://reflectoring.io/100-percent-test-coverage/
+    private EvictingQueue<Double> periodOfValues;
```

Comment:
With an `EvictingQueue`, we are guaranteed that the length of the queue will never exceed the size set at construction time. This is a perfect data structure for maintaining a sliding window of values. 
We must maintain the values in our SMA's period due to the nature of incrementally calculating an SMA. As we'll see later, we need to remove the head's contribution to the mean and then add-in the newly observed value's contribution.